django/db/models/fields/related.py

```diff
@@ -2630,6 +2630,14 @@ def contribute_to_class(self, cls, name, **kwargs):
         # clash.
         if self.rel.symmetrical and (self.rel.to == "self" or self.rel.to == cls._meta.object_name):
             self.rel.related_name = "%s_rel_+" % name
+        else:
+            # If the backwards relation is disabled, replace the original
+            # related_name with one generated from the m2m field name.
+            # This is because Django still uses backwards relations internally
+            # and we need to avoid clashes (multiple m2m fields with
+            # related_name == '+').
+            if self.rel.is_hidden():
```

Comment:
I thought about it and if we use exact match, we would still have the same problem with 2+  arbitrary `related_name`s ending in +.

It seems that we were allowing that to work around a similar bug (https://code.djangoproject.com/ticket/15932) where the developer had to make sure that related names were unique.
This was thought to have ben fixed https://code.djangoproject.com/ticket/21491 but it's now back in a different way.

I think that having an arbitrary `related_name` ending in '+' doesn't add anything as a developer still can't use the backwards relation in their code so I think it's safer to keep the code as it is now as it fixes the bug in (hopefully) all the edge cases.